parachain-tracer/src/tracker.rs

```diff
@@ -521,19 +524,59 @@ impl SubxtTracker {
 		}
 	}
 
-	fn update_core_assignment(
+	async fn get_core_assignments_via_scheduled_paras(
 		&mut self,
-		core_assignments: Vec<polkadot::runtime_types::polkadot_runtime_parachains::scheduler::CoreAssignment>,
-	) {
-		if let Some(index) = core_assignments
+		block_hash: H256,
+	) -> color_eyre::Result<Option<HashMap<u32, Vec<u32>>>> {
```

Comment:
I suggest we get rid of the Option. Just match errors at upper level where we can decide  to try to fetch `ParaScheduler,ClaimQueue` instead